### PR TITLE
35

### PR DESCRIPTION
**Pull request by LewisWelch94**

### CODE REVIEW

File: src/Script/ticketController.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-$(document).ready(() => { loadSummary(); loadProgress(); })
+$(document).ready(() => { loadSummary(); loadProgress(); NavbarActivater.projectActiveTab(); })
```
Comment (OnikNoor97) [inline src/Script/ticketController.js:1]: This Script file will be made redundant, make a new ticket.js file in the public folder and add it is, remember, all JS files will be removed in the future and the new ones will replace it

File: src/Script/projectController.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 $(document).ready(function() 
 {
     loadFeatures();
+    NavbarActivater.projectActiveTab();
```
Comment (OnikNoor97) [inline src/Script/projectController.js:4]: Once again, this Js file will be made redundant

File: src/Script/navBar.js [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
+    projectActiveTab()
+    {
+        document.getElementById("projectNav").classList.add("active");
+    }
+    accountActiveTab()
+    {
+        document.getElementById("accountNav").classList.add("active");
+    }
+}
+NavbarActivater = new Navbar();
```
Comment (OnikNoor97) [inline src/Script/navBar.js:24]: No need to make an instance for NavBar, make the methods static

File: src/Script/navBar.js [hunk 4]
```diff
@@ -1,13 +1,27 @@
...
-    : (navBarActive == "projectPage" || navBarActive == "ticketPage") ? (document.getElementById("projectNav").classList.add("active"))
-        : (navBarActive == "registerPage" || navBarActive == "loginPage" || navBarActive == "adminPage" || navBarActive == "managerPage") ? (document.getElementById("accountNav").classList.add("active"))
-            : null;
+class Navbar
+{
+    constructor()
+    {
 
-$('#searchBarInput').on("keypress", (e) => { if (e.keyCode == 13) searchBar(); });
+    }
```
[the remark this answers is not in the document]
Comment (LewisWelch94) [reply src/Script/navBar.js:10]: fixed

File: src/User/login.php [hunk 5]
```diff
@@ -47,5 +47,6 @@
 	</main>
 <?php include("../../includes/scripts.php"); ?>
 </body>
+<script>NavbarActivater.accountActiveTab();</script>
```
[the remark this answers is not in the document]
Comment (LewisWelch94) [reply src/User/login.php:50]: fixed

File: src/Script/public/register.js [hunk 6]
```diff
@@ -33,4 +33,6 @@ $("#usernameRegister").keyup(async function() {
         $("#registerBtn").prop("disabled", false);
     }
 
-})
\ No newline at end of file
+})
+
+NavbarActivater.accountActiveTab();
```
[the remark this answers is not in the document]
Comment (LewisWelch94) [reply src/Script/public/register.js:38]: fixed

File: src/Script/ticketController.js (hunk 1 above)
Comment (LewisWelch94) [reply src/Script/ticketController.js:1]: fixed

File: src/Admin/index.php [hunk 7]
```diff
@@ -39,6 +39,7 @@
     <?php include("../../includes/editUserModal.php"); ?>
     <?php include("adminModal.php"); ?>
     <script type="text/javascript" src="../Script/admin.js"></script>
+    <script type="text/javascript" src="../Script/public/admin.js"></script>
```
Comment (OnikNoor97) [inline src/Admin/index.php:42]: Do script type="module" for all as eventually every single JS file will use modules

File: src/Home/index.php [hunk 8]
```diff
@@ -20,5 +20,6 @@
         <?php include("../../includes/editUserModal.php"); ?>
         <?php include("../../includes/scripts.php"); ?>
         <script type="text/javascript" src="../Script/home.js"></script>
+        <script type="text/javascript" src="../Script/public/home.js"></script>
```
Comment (OnikNoor97) [inline src/Home/index.php:23]: Same above

File: src/Manager/index.php [hunk 9]
```diff
@@ -32,6 +32,7 @@
     <?php include("../../includes/scripts.php"); ?>
     <?php include("managerModal.php"); ?>
     <script type="text/javascript" src="../Script/managerController.js"></script>
+    <script type="text/javascript" src="../Script/public/manager.js"></script>
```
Comment (OnikNoor97) [inline src/Manager/index.php:35]: Same above

File: src/Project/index.php [hunk 10]
```diff
@@ -60,6 +60,7 @@
 
     <?php include("../../includes/scripts.php"); ?>
     <script type="text/javascript" src="../Script/projectController.js"></script>
+    <script type="text/javascript" src="../Script/public/project.js"></script>
```
Comment (OnikNoor97) [inline src/Project/index.php:63]: Same  above

File: src/Script/navBar.js [hunk 11]
```diff
@@ -1,14 +1,23 @@
...
-        : (navBarActive == "registerPage" || navBarActive == "loginPage" || navBarActive == "adminPage" || navBarActive == "managerPage") ? (document.getElementById("accountNav").classList.add("active"))
-            : null;
-
 $('#searchBarInput').on("keypress", (e) => { if (e.keyCode == 13) searchBar(); });
 
+class Navbar
+{
+     static homeActiveTab()
+    {
+        document.getElementById("homeNav").classList.add("active");
```
Comment (OnikNoor97) [inline src/Script/navBar.js:9]: Use same line curly start brackets as it is JS style to do so 
e.g.
static homeActiveTab() {
// blah blah 
}

File: src/Ticket/index.php [hunk 12]
```diff
@@ -68,6 +68,7 @@
 	<script type="text/javascript" src="../Script/commentController.js"></script>
 	<script type="text/javascript" src="../Script/dateController.js"></script>
 	<script type="text/javascript" src="../Script/ticketController.js"></script>
+	<script type="text/javascript" src="../Script/public/ticket.js"></script>
```
Comment (OnikNoor97) [inline src/Ticket/index.php:71]: Same as above

**Review by OnikNoor97 — CHANGES_REQUESTED** [review]

**Review by OnikNoor97 — APPROVED** [review]

**Pull request merged**